Reservation list

pimber opened the pull request.

pimber committed:
Starting on the Reservation List

pimber committed:
Reservationlist is done

- The CSS have been modifyed and if you don't like it then change it your self plz
    - And some minior modifications in other CSS files i think? Don't no
- The searchbar has been modifyed to search after Name, Stande, and "ej betalt" and "betalt"
- ReservationListItem has been moved to Shared folder
- SurveyPrompt.razor has been deleted
- BookingList.razor has been deleted

pimber committed:
Reservation list

- Have deleted the unused variable in Booking.razor line 15

pimber committed:
ReservationOverview

- Har ændret navnet fra ReservationListItem.razor til ReservationOverview.razor

jonasKjellerup commented on Flea/Pages/Booking.razor:
```diff
@@ -1,6 +1,98 @@
+    };
+    // ---
+    
+    private List<Reservation> _shownReservation = new();
```
The type here can be made more general (`IEnumerable<Reservation>`) allowing for simplification in the search function.

pimber replied:
Fixed

jonasKjellerup commented on Flea/Pages/Booking.razor:
```diff
@@ -1,6 +1,98 @@
+    
+    private List<Reservation> _reservations = new()
+    {
+        {new Reservation() {Name = "Sebastian Aaholm", Stande = 3, PaymentStatus = false}},
```
You don't need the second set of braces.
```cs
new Reservation {Name = "Sebastian Aaholm", Stande = 3, PaymentStatus = false},
```
is sufficient

pimber replied:
Noted

jonasKjellerup commented on Flea/Pages/Booking.razor:
```diff
@@ -1,6 +1,98 @@
+    class Reservation
+    {
+        public string Name { get; set; }
+        public int Stande { get; set; }
```
Avoid using danish words in identifier names. (I am aware that it is merely placeholder but try keep it in mind still)

pimber replied:
Fixed

jonasKjellerup commented on Flea/Pages/Booking.razor:
```diff
@@ -1,6 +1,98 @@
+                in _reservations
+                where c.Stande.ToString().Contains(input)
+                select c;
+            _shownReservation= results.ToList();
```
By implementing the aforementioned change to `_shownReservation` (changing its type to `IEnumerable<Reservation>`) you can simplify this by omitting the intermediary variable and assigning directly to `_shownReservation`.

```cs
_shownReservation = from c
                in _reservations
                where c.Stande.ToString().Contains(input)
                select c;
```

pimber replied:
Fixed thanks for help

jonasKjellerup commented on Flea/Pages/Booking.razor:
```diff
@@ -1,6 +1,98 @@
+                select c;
+            _shownReservation= results.ToList();
+        }
+        else if (input.Contains("betalt"))
```
This case doesn't really make sense unless we also support the inverse case ("ej betalt").

jonasKjellerup commented on Flea/Shared/ReservationOverview.razor:
```diff
@@ -0,0 +1,34 @@
+    public string Name { get; set; }
+
+    [Parameter]
+    public int Stande { get; set; }
```
Again avoid danish words in source code identifiers.

pimber replied:
fixed

jonasKjellerup commented on Flea/Shared/ReservationOverview.razor:
```diff
@@ -0,0 +1,34 @@
+    <button class="button res-del-button" @onclick="Delete">Slet</button>
+    <button class="button res-edit-button" @onclick="Edit">Rediger</button>
+</div>
+
```
The file/component name is slightly misleading as it implies that component implements an overview of reservations, however this is in actuality done by the `Booking` view. This component merely an item in a overview reservation.

jonasKjellerup replied:
This seems to have been solved in 446f8e0

pimber committed:
ReservationOverview page nameing has been fixed

- Flea/Shared/(Reservation.razor) -> (ReservationUserListItem.razor)
- Flea/Pages/ReservationOver.razor has been created and all the code from booking has been transfered to this new page.

jonasKjellerup reviewed (changes requested):
Another more general thing that I think needs to be discussed before merging any more UI related PRs is the approach to styling. We originally discussed trying to make it look as close to native IOS components as possible, hence the inclusion of OnsenUI, however we seem to deliberately be working in opposition to this idea in many cases (e.g the buttons in this commit are using custom CSS override the OnsenUI CSS).

jonasKjellerup committed:
Update css to resemble iOS stylings more

pimber committed:
I dont know what i am doing/remotes/origin/ReservationList' into ReservationList

pimber committed:
Merge remote-tracking branch 'origin/ios_ui' into ReservationList

pimber committed:
ReservationOverview fikset små ændringer

- Se mærsk kommentar for hvad der er ændret
